lib/bolt/transport/base.rb

```diff
@@ -196,6 +196,9 @@ def upload(*_args)
       # This works on deeply nested data structures composed of Hashes, Arrays, and
       # and plain-old data types (int, string, etc).
       def unwrap_sensitive_args(arguments)
+        # Skip this if Puppet isn't loaded
+        return arguments unless defined?(Puppet::Pops::Types::PSensitiveType::Sensitive)
```

Comment:
It might make sense to refactor this later so that we use a Bolt-native type to hide sensitive values.